BLSQ/iaso
# IA-2774 add payment edition modal

Explain what problem this PR is resolving

Related JIRA tickets : IA-2774

## Self proofreading checklist


## Doc
In the code (jsdocs)

## Changes

- Add `useMarkPaymentAsSent` hook to handle the patch request, so the "edit" part of `useSavePaymentLot` can be used to update payments' statuses in the lot
- Add separate modal for edition with individual buttons to edit payment lot fields and without formik or yup validation
- Use value from table row to populate table and get initial state value
- update `patchRequest` of `useSavePaymentLots`: commented out actual patch request (already formatted for use with API). log body in the console for the time being
- use `patchRequest` in `PaymentLotEditionDialog` to create callbacks for "save name "and "save comment" buttons
- Name and comment are already updated using the payment lot API (no mock)
- Refactor `usePotentialPaymentsColumns` -> `usePaymentsColumns`, with a `potential` prop
- Add `useTableSelection` convenience hook
- Add `EditPaymentDialog`:
    - Update a single table row
    - API mocked at the moment 
- Add `BulkEditPaymentDialog`:
    - Update selection of rows
    - Add custom "select all" and "unselect all" buttons 
    - Add warning modal on confirm. Layout taken from org unit bulk update, but the underlying component is `ConfirmCancelDialog`

## How to test

- Create a payment lot
- Go to Payment lots
- Click the edit icon button
- Test the different fields/buttons:
    - Name and comment already call the backend so they will be updated
    - Try editing a single payment 
    - Try selecting/unselecting multiple payments, then saving them
        - There should be a warning modal with the number of payments about to be updated
        - There should be a log of the body of the POST request in the console 

## Print screen / video


https://github.com/BLSQ/iaso/assets/38907762/05aea426-cbaf-4a6c-8b30-d4f606e62740


## Notes

Has to be merged after IA-2791, as it already implements it. PR: https://github.com/BLSQ/iaso/pull/1184
